src/components/Link.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <RouterLink
+    v-if="useRouterLink"
+    class="ae-link"
+    :to="to"
+  >
+    <slot />
+  </RouterLink>
+  <a
+    v-else
+    class="ae-link"
+    :href="to"
+    :target="isLinkOnSameHost ? '_self' : '_blank'"
+    @click="$emit('click', $event)"
+  >
+    <slot />
+  </a>
+</template>
+
+<script>
+export default {
+  props: {
+    to: { type: [String, Object], required: true },
+  },
+  computed: {
+    isLinkOnSameHost() {
+      return typeof this.to === 'object'
+        || (new URL(this.to, window.location)).host === window.location.host;
+    },
+    useRouterLink() {
+      return this.$options.components.RouterLink && this.isLinkOnSameHost;
```

Comment:
This was needed to be able to use it without VueRoute, it can be removed in this project